packages/drawer/src/react/collapsible.js

```diff
@@ -0,0 +1,101 @@
+import core from '@pluralsight/ps-design-system-core'
+import glamorous from 'glamorous'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+const Container = glamorous.div({
+  overflow: 'hidden',
+  transition: `height ${core.motion.speedNormal}`,
+})
+
+export default class Collapsible extends React.Component {
+  componentDidMount() {
+    this.updateOverflowStyle(this.props.isOpen)
+    if (!this.props.isOpen) {
+      this.containerElement.style.height = 0
+    }
+  }
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      this.toggle(nextProps.isOpen)
+    }
+  }
+
+  toggle(isOpen) {
```

Comment:
The toggle, open, and close stuff is complicated!  I'm glad you brought your experience and time with this problem to help the greater system.  This is exactly the kind of stuff that we would have missed without your help! Thank you.

Reply:
In fact, I would love to just sit with you and have you verbally walk through some of these lines so I can understand it better and what it helps solve.  Would that be ok?

Reply:
done.  thanks for the extra info.